siliconcompiler/core.py

```diff
@@ -398,6 +398,18 @@ def post_process(cmdargs):
         except ValueError as e:
             self.error(f'{e}', fatal=True)
 
+    def set_dependency(self, name, path, ref):
```

Comment:
maybe make this `register_package_source` and add docstring
also add `clobber=True` as parameter